frontend/src/libs/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,152 @@
+import { Button, IconButton } from '~/libs/components/components.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback, useMemo } from '~/libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+const ONE_STEP_LENGTH = 1;
+const MAX_VISIBLE_PAGES = 3;
+const FIRST_PAGE = 1;
+
+type Properties = {
+  currentPage: number;
+  onChangePage: (value: number) => void;
+  count: number;
+  rowsPerPage: number;
+  className?: string | undefined;
+};
+
+const Pagination: React.FC<Properties> = ({
+  currentPage,
+  onChangePage,
+  count,
+  rowsPerPage,
+  className = '',
+}: Properties) => {
+  const handlePreviousPage = useCallback(() => {
+    onChangePage(currentPage - ONE_STEP_LENGTH);
+  }, [onChangePage, currentPage]);
+
+  const handleNextPage = useCallback(() => {
+    onChangePage(currentPage + ONE_STEP_LENGTH);
+  }, [onChangePage, currentPage]);
+
+  const handleChangePage = useCallback(
+    (page: number) => {
+      return () => {
+        onChangePage(page);
+      };
+    },
+    [onChangePage],
+  );
+
+  const totalPages = useMemo(() => {
+    return Math.ceil(count / rowsPerPage);
+  }, [count, rowsPerPage]);
+
+  const visiblePages = useMemo(() => {
+    const pages = [];
+    for (let page = currentPage; page <= totalPages; page++) {
+      if (pages.length === 0) {
+        if (page === FIRST_PAGE) {
+          pages.push(page);
+        } else if (page === totalPages) {
+          if (page === FIRST_PAGE + ONE_STEP_LENGTH) {
+            pages.push(page - ONE_STEP_LENGTH, page);
+          } else {
+            pages.push(
+              page - 2 * ONE_STEP_LENGTH,
+              page - ONE_STEP_LENGTH,
+              page,
+            );
+          }
+        } else {
+          pages.push(page - ONE_STEP_LENGTH, page);
+        }
+        continue;
+      }
+
+      if (pages.length === MAX_VISIBLE_PAGES) {
+        break;
+      }
+      pages.push(page);
```

Comment:
Can't we achieve this with slice method?

Reply:
Achieve code with switch-case and filter method